upgrading it to v0.2.3

jing2uo opened the pull request:
update from https://longbridge.com/desktop/

jing2uo committed:
Update to v0.2.3.

Signed-off-by: jing guo <mail@guojing.io>

hfiguiere commented:
so what happens to the aarch64 package now?

hfiguiere commented on com.longbridgeapp.LongbridgePro.yaml:
```diff
@@ -1,23 +1,23 @@
-base-version: '23.08'
-command: longbridgepro
+runtime: org.gnome.Platform
+runtime-version: "48"
```
why does it want the GNOME 48 runtime now?

jing2uo replied:
need  libwebkit2gtk-4.1-0

jing2uo commented:
> so what happens to the aarch64 package now?

The previous application stopped development for a long time. Recently, they released a new Rust-based program, which does not provide an aarch64 version. You can view the release page here: https://longbridge.com/desktop/release-notes/v0.2.0

hfiguiere commented:
nonetheless a broken aarch64 package will be built.

hfiguiere commented on com.longbridgeapp.LongbridgePro.yaml:
```diff
@@ -27,50 +27,21 @@ modules:
-          type: anitya
-          project-id: 20333
-          url-template: https://www.openssl.org/source/openssl-$version.tar.gz
+        url: https://assets.lbkrs.com/github/release/longbridge-desktop/stable/longbridge-v0.2.1-linux-x86_64.deb https://assets.lbctrl.com/github/release/longbridge-desktop/stable/longbridge-v0.2.0-linux-x86_64.deb
```
this URL is invalid

jing2uo replied:
sorry..

jing2uo commented:
> nonetheless a broken aarch64 package will be built.

Is there any way to solve this?

hfiguiere commented:
it's in the documentation https://docs.flathub.org/docs/for-app-authors/maintenance#limiting-the-set-of-architectures-to-build-on

jing2uo commented:
> it's in the documentation https://docs.flathub.org/docs/for-app-authors/maintenance#limiting-the-set-of-architectures-to-build-on
There is a configuration for only-arches here:
https://github.com/flathub/com.longbridgeapp.LongbridgePro/pull/1/files#diff-a0f24540baae8f0c09e0f50a22e0f4c129e67b0db30f04cd27ceb9f46663c400R44

hfiguiere commented:
did you read the doc?

jing2uo commented:
> did you read the doc?

Sorry, I didn't correctly understand the meaning of the documentation earlier.

jactry commented:
Hi @jing2uo ,

Could you combine these commits into one commit? Since they are all for the one purpose - upgrading it to v0.2.1.

Thanks for working on this!

jing2uo commented:
> Hi @jing2uo ,
> 
> Could you combine these commits into one commit? Since they are all for the one purpose - upgrading it to v0.2.1.
> 
> Thanks for working on this!

done!

jactry commented:
Hi @jing2uo , did you encounter this issue? All the Chinese characters (Simplified or Traditional) aren't displayed. The English interface is fine.

![longbridge_chinese](https://github.com/user-attachments/assets/9a7cd2e6-1a65-41ca-be10-975ba20cd068)

jing2uo commented:
> Hi @jing2uo , did you encounter this issue? All the Chinese characters (Simplified or Traditional) aren't displayed. The English interface is fine.
> 
> ![longbridge_chinese](https://private-user-images.githubusercontent.com/1217301/442397531-9a7cd2e6-1a65-41ca-be10-975ba20cd068.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY5MTczOTgsIm5iZiI6MTc0NjkxNzA5OCwicGF0aCI6Ii8xMjE3MzAxLzQ0MjM5NzUzMS05YTdjZDJlNi0xYTY1LTQxY2EtYmUxMC05NzViYTIwY2QwNjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTBUMjI0NDU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTY0ZWI4Y2VlMGJjYjJkNjBkN2E3ZTMyOWNkMGI1ZWJkYmIwMDBmNmYwODQ5ZWMwNDNkNjU4YzA1ZTYyOTI2NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cwdBsPi-rRIi8teBfMEIVAFsaZ86tcGmtT7EdRterio)

My environment works fine. I tested it on Debian Sid and Arch without encountering this issue. Chinese is my default display language, and the noto-cjk font is installed in both. Can you try running the deb or AppImage from the official website in your environment to check for issues? In fact, the deb package contains only one binary: longbridge, which you can find in the `.local/share/flatpak/app/com.longbridgeapp.LongbridgePro/x86_64/test/active/files/extra/` directory. It‘s depends on the external environment's libwebkit2gtk.

jactry commented:
On 5/11/25 00:04, Komh wrote:
> Can you try running the deb or AppImage from the official website in your environment to check for issues?

Yeah, both the AppImage and the binary from the .deb package work well 
on my Debian testing (with fonts-noto-cjk and fonts-noto-cjk-extra 
installed), even with an en_US.UTF-8 locale.
Let me give it a try with a fresh installation system.

jing2uo commented:
Asked a few friends to help test, and the issue can indeed be reproduced in their environments. I suspect it’s caused by fontconfig, but deleting my own configuration still allows normal display. I’ll find time to set up a clean environment for further testing. @jactry

jing2uo commented:
@jactry 

Please check the latest build to see if the Chinese font issue has been resolved.


flatpak install --user https://dl.flathub.org/build-repo/188571/com.longbridgeapp.LongbridgePro.flatpakref


 I noticed that `~/.local/share/flatpak/runtime/org.gnome.Platform/x86_64/48/active/files/share/fonts` (I’ve been using `flatpak --user`) contains a large number of fonts. When I deleted them and added the Noto fonts, the Longbridge interface displayed correctly. So, I added a `fonts.conf` file and set the `FONTCONFIG_FILE` environment variable to override the fonts provided by the GNOME runtime, and it works fine in my environment. Looking forward to your response.

jing2uo commented:
@jactry 
Please verify if the latest build resolves the issue of Chinese characters not displaying correctly.

I observed that ~/.local/share/flatpak/runtime/org.gnome.Platform/x86_64/48/active/files/share/fonts (using flatpak --user) contains numerous fonts. After removing them and adding Noto fonts to the directory, the Longbridge interface displayed correctly. The issue arises because the application match fonts from the flatpak gnome runtime that cannot properly render Chinese characters. To resolve this, I added a fonts.conf file and set the FONTCONFIG_FILE environment variable to override the fonts in runtime, which fixed the issue in my environment. 

Looking forward to your feedback. Please grant me project maintenance permissions if possible, and I’ll actively update it to address any further issues. Thank you!

jing2uo commented:
@jactry   Apologies for the interruption, please continue reviewing and testing this PR.

jactry commented:
Hi,

> @jactry Apologies for the interruption, please continue reviewing and testing this PR.

So sorry for the delay! I was always keeping an eye on your progress in this thread, I just didn't find time to take a close look at it and test it. I will finish it within this week.

Thanks

jactry commented on com.longbridgeapp.LongbridgePro.yaml:
```diff
@@ -1,36 +1,38 @@
-  - --socket=x11
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --env=FONTCONFIG_FILE=/app/share/fonts/fonts.conf
```
I can confirm this works for me.
I am still curious why we need this for Longbridge while other Flatpak applications don't require this to work (I found nothing by searching `FONTCONFIG_FILE` in the FlatHub organization), is it because the application is hard-coding something? Or just because Longbridge triggers a Flatpak bug?
I see Longbridge also provides a .AppImage which includes freetype and some other libraries. I would like to try to generate a Flatpak from it and use the libraries it ships instead, to see if there are any differences to .deb.

jing2uo replied:
Could you please share any progress on researching the font display issue for the AppImage version?

jing2uo replied:
Maybe we can merge first and remove the special configuration of FONTCONFIG_FILE later after resolving the issues? FONTCONFIG_FILE doesn't introduce security issues or bugs,  and it doesn't affect user experience either. The old version hasn't been updated in over a year, while the new version iterates quickly. If users prefer the old version, they can choose to lock it and not update.

jactry commented on com.longbridgeapp.LongbridgePro.yaml:
```diff
@@ -1,36 +1,38 @@
-          - mv opt/Longbridge\ Pro longbridgepro
-          - rm -r longbridgepro.deb opt usr
+          - mv usr/local/bin/longbridge longbridge
+          - rm -rf longbridgepro.deb opt usr
```
Let's go with `- rm -r longbridgepro.deb usr` instead.

jactry commented on com.longbridgeapp.LongbridgePro.metainfo.xml:
```diff
@@ -2,17 +2,17 @@
   <id>com.longbridgeapp.LongbridgePro</id>
   <metadata_license>CC0-1.0</metadata_license>
-  <project_license>LicenseRef-proprietary=https://support.longbridgeapp.com/topics/us-trade/user-agreement</project_license>
+  <project_license>LicenseRef-proprietary=https://longbridge.com/hk/en/support/topics/misc/3isdzoe</project_license>
```
Any reason to use this instead of the original one? Is there a reference from the new version of Longbridge?

jactry reviewed (commented):
Hi Jing Guo,

Sorry again for my long delay, thanks for your patience!

Some comments below. Other than that, I am also considering that we should push this to a beta branch first, at least before we figure out the root of the font issue, since the new version of the application is developed in a totally different framework, and meanwhile, the old version still works. Does this sound reasonable to you?

jing2uo commented:
> Some comments below. Other than that, I am also considering that we should push this to a beta branch first, at least before we figure out the root of the font issue, since the new version of the application is developed in a totally different framework, and meanwhile, the old version still works. Does this sound reasonable to you?

ok!
